### ISSUE TITLE
Singular+plural terms in TD are confusing

### ISSUE THREAD

**Issue by zolkis** [w3c/wot-thing-description#359]
Currently [Thing](https://w3c.github.io/wot-thing-description/#thing) has _description_, _descriptions_, [InteractionPattern](https://w3c.github.io/wot-thing-description/#interactionpattern) has these plus _title_ and _titles_, etc.

I heard these are because e.g. _description_ is in the default document language and _descriptions_ is an array with descriptions in other languages.

This is rather confusing. Instead, why not have a (DOMString or [MultiLanguage](https://w3c.github.io/wot-thing-description/#multilanguage)) array _description_  which is either a single string or an array where the first entry is in the document default language?

BTW "Container" in 5.2.10 (MultiLanguage definition) is not exact. Examples would also be needed.

**Comment by sebastiankb** [comment]
I would like to refer to this issue here https://github.com/w3c/wot-thing-description/issues/161
This should answer your questions.

**Comment by zolkis** [comment]
If singular+plural terms are for "solving" the language problem, then it's blowing a non-problem into a solution, and a solution that is not even needed.

In that issue I agree with the [comment](https://github.com/w3c/wot-thing-description/issues/161#issuecomment-408101127) of @benfrancis that language negotiation should precede fetching the TD.

It is unnecessary and inefficient to carry possibly hundreds of language definition for all affected names in a TD when a given client will mostly need one, and when needing more, will fetch again with a different accept-language header.

Either way, having both `title` and `titles` etc remains confusing. Probably `titles` would need a name change at least. But I'd rather solve the language issue in a better way.

**Comment by sebastiankb** [comment]
I'm not very sure if your read the full [thread](https://github.com/w3c/wot-thing-description/issues/161) and read the responses of @benfrancis comment. In addition you should also check the minutes of the TD web meetings at that time.

>It is unnecessary and inefficient to carry possibly hundreds of language definition for all affected names in a TD when a given client will mostly need one, and when needing more, will fetch again with a different accept-language header.

You don't need. As servient, you still can work with content negation and offer TDs to clients based on its (language) request. This works perfect with HTTP and CoAP. What are you doing with MQTT as example? For such a case (and other use cases) the TD provides the options to offer multi languages information in a single TD.

> Either way, having both title and titles etc remains confusing. Probably titles would need a name change at least. But I'd rather solve the language issue in a better way.

What would be your proposal?

**Comment by sebastiankb** [comment]
>I would use single names like title, description etc, with type of either (string or array). If the provided value is a string, it is assumed to be in the document default language. If the provided value is an array, the first entry is assumed to be in the document default language, and the rest are in other languages.

We also evaluated this. However, this would be not JSON Schema compatible anymore. 

Btw: I will add a note in the TD spec to mention that content negation may be used to announce the preferred language.

**Comment by egekorkan** [comment]
Princeton Testfest:
No one has voted against having this note.

**Comment by egekorkan** [comment]
From Princeton Testfest:

@zolkis Can you please change the name of this issue which mentions the multiple languages. This issue is specifically meant for multi languages

**Comment by zolkis** [comment]
Hmm, the issue I had was the mix of singular and plural terms that I have found confusing - the title reflects that.

The reason given for this was the one of multiple languages, but that doesn't make this issue a language issue: it is still a vocabulary/semantics issue. I have found no other spec with this kind of "solution". We should try to make a solution to the multi-language problem in a way that does not involve name and name+s properties.
